symfony/plugins/orangehrmLeavePlugin/Service/LeaveAssignmentService.php

```diff
@@ -122,73 +128,71 @@ protected function saveLeaveRequest(LeaveParameterObject $leaveAssignmentData) {
+                    $loggedInEmpNumber = $this->getAuthUser()->getEmpNumber();
+                    $leaveRequest = $this->getLeaveRequestService()
+                        ->getLeaveRequestDao()
+                        ->saveLeaveRequest($leaveRequest, $leaveDays, $entitlements);
+
+                    if (!empty($leaveAssignmentData->getComment())) {
+                        $leaveRequestComment = new LeaveRequestComment();
+                        $leaveRequestComment->setLeaveRequest($leaveRequest);
+                        $leaveRequestComment->setCreatedAt(new DateTime());
+                        $leaveRequestComment->getDecorator()->setCreatedByUserById($loggedInUserId);
+                        $leaveRequestComment->getDecorator()->setCreatedByEmployeeByEmpNumber($loggedInEmpNumber);
+                        $leaveRequestComment->setComment($leaveAssignmentData->getComment());
+                        $this->getLeaveRequestService()
+                            ->getLeaveRequestDao()
+                            ->saveLeaveRequestComment($leaveRequestComment);
                     }
 
-                    /* Send notification to the when leave is assigned; TODO: Move to action? */
-                    
-                    /** TODO: Don't hard code 'ADMIN' here: Use method to get allowed actions ordered by priority and choose the assign action */
-                    
-                    $workFlow = $this->getWorkflowService()
-                             ->getWorkflowItemByStateActionAndRole(WorkflowStateMachine::FLOW_LEAVE, 'INITIAL', 'ASSIGN', 'ADMIN');
-                                        
-                    $eventData = array('request' => $leaveRequest, 'days' => $leaveDays, 'empNumber' => $_SESSION['empNumber'],
-                        'workFlow' => $workFlow);
-                    $this->getDispatcher()->notify(new sfEvent($this, LeaveEvents::LEAVE_CHANGE, $eventData));
+//                    /* Send notification to the when leave is assigned; TODO: Move to action? */
+//
+//                    /** TODO: Don't hard code 'ADMIN' here: Use method to get allowed actions ordered by priority and choose the assign action */
+//
+//                    $workFlow = $this->getWorkflowService()
+//                             ->getWorkflowItemByStateActionAndRole(WorkflowStateMachine::FLOW_LEAVE, 'INITIAL', 'ASSIGN', 'ADMIN');
+//
+//                    $eventData = array('request' => $leaveRequest, 'days' => $leaveDays, 'empNumber' => $_SESSION['empNumber'],
+//                        'workFlow' => $workFlow);
+//                    $this->getDispatcher()->notify(new sfEvent($this, LeaveEvents::LEAVE_CHANGE, $eventData));
 
 //                    return true;
                     return $leaveRequest;
-                } catch (Exception $e) {
-                    throw new LeaveAllocationServiceException('Error saving leave request');
+                } catch (\Exception $e) {
```

Comment:
format the code with `Optimize Import` tick

![image](https://user-images.githubusercontent.com/25487187/134453616-c7da8fd1-dcdc-4e08-9fc8-3ee3a52d1342.png)

Reply:
Changed into Exception instead of \Exception .